### ISSUE TITLE
is_dataclass always evaluates to false with TypeVars

### ISSUE THREAD

**Issue by peku33** [microsoft/pyright#8686]
`dataclasses.is_dataclass` seems to always evaluate to false for generic types.

<h3> Environment data </h3>

-   Language Server version: 2024.8.1
-   OS and version: darwin arm64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.11.9
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: strict

<h3> Code Snippet </h3>

```python
from dataclasses import is_dataclass
from typing import TypeVar

T = TypeVar("T")


def test_fn(type_: type[T]) -> T:
    assert is_dataclass(type_)

    print("this is marked as unreachable")

    raise NotImplementedError()

```

**Comment by peku33** [comment]
It looks like this has been fixed in #8738

**Comment by erictraut** [comment]
This still needs to be fixed.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.380.
